### PR TITLE
Dot

### PR DESCRIPTION
**Pull request by maxerogers**
https://github.com/olado/doT 
http://olado.github.io/doT/

- include in pkg.json
- implement doT adapter
- include doT tests

:rowboat:

### CODE REVIEW

**Comment by kylemac** [comment]
i can verify that these tests passed locally just fine, but we'll get to the bottom of the :red_circle: in the morning

File: test/test.coffee [hunk 1]
```diff
@@ -323,6 +323,71 @@ describe 'stylus', ->
...
+  # dot doesn't support external file requests out of the box. You have to write your own extension to load snippets.
+  # try using the one found here https://github.com/olado/doT/blob/master/examples/withdoT.js
+  it 'should handle partial renders', (done) ->
+    lpath = path.join(@path, 'partial.dot')
+    @dot.renderFile(lpath, { name:"Jake",age:31})
+      .done((res) => should.match_expected(@dot, res.result, lpath, done))
+
+  it 'should client-compile a string', (done) ->
+    @dot.compileClient("{{? it.name }}
+<div>Oh, I love your name, {{=it.name}}!</div>
```
Comment (slang800) [inline test/test.coffee]: this section should prolly be indented... maybe using a multiline string?

File: test/fixtures/haml/expected/rstring.html [hunk 2]
```diff
@@ -1,2 +1,2 @@
 
-<div class="foo">Whats up mang</div>
\ No newline at end of file
+<div class="foo">Whats up mang</div>
```
Comment (slang800) [inline test/fixtures/haml/expected/rstring.html]: this doesn't look like it belongs in this PR

File: readme.md [hunk 3]
```diff
@@ -93,11 +93,13 @@ It's also important to note that accord returns an object rather than a string f
 - [swig](http://paularmstrong.github.io/swig)
 - [marc](https://github.com/bredele/marc)
 - [toffee](https://github.com/malgorithms/toffee)
+- [doT.js](https://github.com/olado/doT)
 - nunjucks _(pending)_
 - haml-coffee _(pending)_
 - dust _(pending)_
 - underscore _(pending)_
 
+
```
Comment (slang800) [inline readme.md]: little extra whitespace

File: package.json [hunk 4]
```diff
@@ -51,7 +51,8 @@
     "react-tools": "latest",
     "stylus": "latest",
     "swig": "latest",
-    "toffee": "latest"
+    "toffee": "latest",
+    "dot": "latest"
```
Comment (slang800) [inline package.json]: if you run [fixpack](https://www.npmjs.com/package/fixpack) on this, it'll automatically alphabetize it for you... I think `npm install dot --save-dev` will do the same.

**Comment by jenius** [comment]
Looking good so far! Slang hit all the concerns I had above :+1:

**Comment by jenius** [comment]
Looks good :+1:

**Comment by jenius** [comment]
Missing a docs entry, and would love for e126dbb to be squashed into 9c274dd, then the whole thing to be rebased on top of master. After this, should be ready to merge!

**Comment by maxerogers** [comment]
Removed the pending

**Comment by jenius** [comment]
:pear:-fecto

**Comment by jenius** [comment]
Thanks @maxerogers for the good work here! :tada:

**Comment by maxerogers** [comment]
:bowtie:  anytime @jenius

**Pull request merged**